src/tox/venv.py

```diff
@@ -223,7 +223,7 @@ def is_allowed_external(self, p):
         if tox.INFO.IS_WIN:
             tryadd += [os.path.normcase(x) for x in os.environ["PATHEXT"].split(os.pathsep)]
             p = py.path.local(os.path.normcase(str(p)))
-        for x in self.envconfig.whitelist_externals:
+        for x in self.envconfig.allowlist_externals + self.envconfig.whitelist_externals:
```

Comment:
I'm not happy to allow specifying both of them. We should instead allow only one, and raise if both set. And then pick only the one that is set.